models/evc.py

```diff
@@ -148,12 +152,13 @@ def __init__(self, controller, **kwargs):
         self.flow_removed_at = get_time(kwargs.get("flow_removed_at")) or None
         self.updated_at = get_time(kwargs.get("updated_at")) or now()
         self.execution_rounds = kwargs.get("execution_rounds", 0)
+        self.error_status = kwargs.get("error_status", False)
 
         self.current_links_cache = set()
         self.primary_links_cache = set()
         self.backup_links_cache = set()
         self.affected_by_link_at = get_time("0001-01-01T00:00:00")
-        self.old_path = Path([])
+        self.old_paths = [Path([])]
```

Comment:
I'm not convinced this is a good idea to be a list of paths. The old path was meant to only hold the soon-to-deleted prior path which is either the `failover_path` (if it's been affected) or the  old `current_path`. 

Let me know if you had any other major reason to justify this, as it is I don't think we should keep appending old_paths. Like, either a path when sending flows can fail to remove or install flows, it can end up in the error state, which is a good idea, but trying to store multiple or repeated paths, can be very confusing. 

If you were to model the `error_status: bool` as `status_reason: set[str]`, then you could also map the exceptions you introduce `EVCPathNotDeleted` and `EVCPathNotInstalled` with their unique string representation like what the `status_reason` value should be like, `deployed_error`, `deletion_error`, meaning if what we have is `deployed_error` then it needs to be redeployed later to be fixed, if it was just a `deletion_error` that means it was from a EVC deletion operation and couldn't get deleted, so eventually it should be deleted by a user (or if one day consistency check can try to repeat this op here too). So, notice that just mapping the action that has failed whether flow installed failed or flow removed failed might not be enough since the context isn't captured, what matters the most is how to get out of the error situation or what to do next. Think about

Does that make sense to you? Or do you see any other way that can make it better?

Reply:
Yes, I agree `old_paths` can become really heavy and hard to be dealt with. This part can be changed back and the old path can be deleted by event instead of request. Or also just send them through an event after the retries.

The motive for this change was because I caught the `Service Unavailable` when `topology.link_down` was sent and all the installed EVCs were trying to delete `old_path` in addition to re-deployments I trigger concurrently. An error when deleting an old path does not change EVC `error_status` (to be changed to str). So it was just the event `kytos/mef_eline.cleanup_evcs_old_path` being in a loop with EVCs retrying to delete their old paths until `flow_manager` starts to response with 200s.

This part has an added complex since `cleanup_evcs_old_path` now has two triggers and also both modify object lists. Now thinking about it, this process alone is not concurrent so it should be fine with just retries.